homework/index.js

```diff
@@ -29,26 +37,163 @@
     return elem;
   }
 
+  function appendRepoDetail(header, description, parentTable, link) {
+    const repoDetailRow = createAndAppend('tr', parentTable);
+    createAndAppend('th', repoDetailRow, { text: header });
+    const descriptionElm = createAndAppend('td', repoDetailRow);
+    if (link) {
+      createAndAppend('a', descriptionElm, {
+        href: link,
+        text: description,
+        target: '_blank',
+      });
+    } else {
+      descriptionElm.textContent = description;
+    }
+    return repoDetailRow;
+  }
+
+  function getDateTimeText(dateTimeStr) {
+    const dateTime = new Date(dateTimeStr);
+    return dateTime.toLocaleString();
+  }
+
+  function sortRepositoriesByNameAscending(firstRepo, secondRepo) {
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
+  function renderRepositoryNames(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function sortAndMapRepositories(repositoryInfos) {
+    return repositoryInfos
+      .sort(sortRepositoriesByNameAscending)
+      .map(repositoryInfo => ({
+        name: repositoryInfo.name,
+        url: repositoryInfo.html_url,
+        description: repositoryInfo.description,
+        forks: repositoryInfo.forks,
+        lastUpdateTime: getDateTimeText(repositoryInfo.updated_at),
+        contributorsUrl: repositoryInfo.contributors_url,
+      }));
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const repoDetailsListItem = createAndAppend('li', ul, {
+      class: 'repository-item',
+    });
+    const repoDetailsTable = createAndAppend('table', repoDetailsListItem, {
+      class: 'repository-item-table',
+    });
+    appendRepoDetail('Repository:', repo.name, repoDetailsTable, repo.url);
+    appendRepoDetail('Description:', repo.description, repoDetailsTable);
+    appendRepoDetail('Forks:', repo.forks, repoDetailsTable);
+    appendRepoDetail('Updated:', repo.lastUpdateTime, repoDetailsTable);
+  }
+
+  function renderContributorDetails(contributor, ul, index) {
+    const repoContributorListItem = createAndAppend('li', ul, {
+      class: 'contributors-item',
+    });
+    if (index === 0) {
+      createAndAppend('h6', repoContributorListItem, {
+        text: 'Contributions:',
+      });
+    }
+    createAndAppend('img', repoContributorListItem, {
+      src: contributor.avatar_url,
+      alt: 'Contributor avatar picture',
+    });
+    createAndAppend('a', repoContributorListItem, {
```

Comment:
Instead of creating a link here, you could create a link to encloses all the content of the list item. In that way, you could click anywhere on the list item to go to the contributor's page.